subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformTestFixture.groovy

```diff
@@ -76,10 +75,9 @@ class Producer extends DefaultTask {
         buildFile << """
 allprojects {
     dependencies {
-        registerTransform {
+        registerTransformAction(MakeGreen) {
```

Comment:
`registerTransformAction` looks weird. Can we do something like `transform()` instead perhaps? Or `transformWith()` maybe?

Reply:
I would like to keep the `register` in the name, so it is not completely different to the other registrations.

The other option would be to also use `registerTransform` for both kinds and then decide if the type is
- annotated with `@TransformAction` -> it is a parameter object
- of type `ArtifactTransformAction` -> it is a parameterless transform, calls to `parameters` or `getParameters` would fail.

WDYT?

Reply:
I guess we'd need two overloads in this case, too? One to take `Object` the other to take `ArtifactTransformAction`.

There's also the option to go with `registerTransformWithParameters()`. Or `registerParameterizedTransform()`.

Reply:
We could accept whatever is in this PR for now, and then add this name to the list of names we want to decide on.